### DISCUSSION TITLE
Initital Setup - Stuck in Unknown Phase?

### DISCUSSION THREAD

**Discussion by AJCastroChandri** [blakeblackshear/frigate#14694] [Ask A Question] [answered]
Hello,
I setup a new computer with debian12 (64gb ram, AMD Ryzen9 with I think 12 cores, and an Nvidia GeForce 3080 gpu).
The web-interface for frigate is slow as hell... and I believe I am getting no detection on the two cameras I setup.

I have tried to follow the instructions and do things by-the-book... yet I still don't understand where I'm failing.

Below I am attaching the logs I obtained via `docker logs frigate`.
I'm still trying to wrap my head around docker, and passing the GPU hardware to it, and detecting the models.

Below attached the logs, my config.yml, and my docker-compose.yml.

Do you see anything wrong?
If not, any additional information I can obtain to help you help me figure t
[logs.txt](https://github.com/user-attachments/files/17577326/logs.txt)
his out?
I thank you... as I'm at a loss.

[logs.txt](https://github.com/user-attachments/files/17577330/logs.txt)
[config.txt](https://github.com/user-attachments/files/17577332/config.txt)
[docker-compose.txt](https://github.com/user-attachments/files/17577333/docker-compose.txt)
EDIT: the .yml files have a ".txt" extension, as github doesn't permit .yml to be attached to a post.

**Comment by NickM-27** [comment, marked as answer]
>       path: /var/lib/docker/overlay2/44b72aca3e66a2401ab478f426b5ad92f946be893cc6f09b50df53d9082d1554/merged/config/model_cache/tensorrt/8.5.3/yolov7-320.trt

this is not the right path, it should just be `/config/model_cache/tensorrt/8.5.3/yolov7-320.trt`

but you also need to define the model properties which you have not done https://docs.frigate.video/configuration/object_detectors#configuration-parameters

**Reply by AJCastroChandri** [reply]
Thank you @NickM-27 !
I modified the config.yml, and now the "detectors stanza" reads like this:
```
detectors:
  nvidia:
    type: tensorrt
    device: 0
    model:
      path: /home/aj/config/model_cache/tensorrt/yolov7-320.trt
      input_tensor: nchw
      input_pixel_format: rgb
      width: 320
      height: 320
```
That model file (yolov7-320.trt), I picked out from the one the system generated, which it had placed inside: 
```Thank you @NickM-27 !
I modified the config.yml, and now the "detectors stanza" reads like this:
```
detectors:
  nvidia:
    type: tensorrt
    device: 0
    model:
      path: /home/aj/config/model_cache/tensorrt/yolov7-320.trt
      input_tensor: nchw
      input_pixel_format: rgb
      width: 320
      height: 320
```
That model file (yolov7-320.trt), I picked out from the one the system generated, which it had placed inside: 
```/var/lib/docker/overlay2/ce7729f54b33cf01133f23e26e937f0ba7d739cd1327f3599b08955506e16432/diff/config/model_cache/tensorrt/8.5.3/yolov7-320.trt
```
directory here --> "/var/lib/docker/overlay2/ce7729f54b33cf01133f23e26e937f0ba7d739cd1327f3599b08955506e16432/diff/config/model_cache/tensorrt/8.5.3/yolov7-320.trt"

I took the file from there and copied it over to /home/aj/

I'm still new to this, yet: every time I do ``` docker-compose down && docker-compose up -d```
I check the logs, and the machine is generating the model again (it says to wait a few minutes).

Would it be something in my config that is telling it to re-generate on every docker down/up?
THANK YOU!


```

I took the file from there and copied it over to /home/aj/

I'm still new to this, yet: every time I do ``` docker-compose down && docker-compose up -d```
I check the logs, and the machine is generating the model again (it says to wait a few minutes).

Would it be something in my config that is telling it to re-generate on every docker down/up?
THANK YOU!

**Reply by AJCastroChandri** [reply]
My god I broke the online wysiwyg editor...

**Comment by AJCastroChandri** [comment]
Scratch my last mess of a message @NickM-27 !
I re-mapped the .rtr, and it is working awesome!!!!!

Lemme play with this and I'll update.
THANK YOU!

**Comment by AJCastroChandri** [comment]
So I've got a few questions... which I'm still trying to wrap my head around.

1. In docker, I have not "mapped" the HDDs present in the server, over to the container.  Once this is done, Frigate would store the videos/events over in those HDDs?   I ask because the server has one m2 for the OS, and three HDDs for "space".
2. I created a user in Frigate via the web interface... yet I did docker down/up, and when it came back up, the user had disappeared.  Is there a place I can "persist" all these configurations?
3. Whenever I do docker down/up, the logs ("docker logs frigate") show that it is generating once again the "model".  Is this normal?  I don't plan on doing docker down/up frequently, yet I'm wondering if I'm doing something wrong.
4. I happily paid the yearly subscription for Frigate+.  I don't know if I'll use it, yet I wanted to show my support somehow.

Thank you!

**Reply by NickM-27** [reply]
1. yes
2. this means your docker config is not correct and the database was not saved correctly
3. no, it means you have something wrong in your docker config

**Comment by AJCastroChandri** [comment]
Ok, so lots of progress here!

Right now, I was able to map the database... now upon running docker down/up, the created users are persistent.

So that is good!
Yet the model file is still getting re-generated on every docker up.

Below the 'tail' of the logs:
`aj@frigate:~$ docker logs -f frigate
s6-rc: info: service s6rc-fdholder: starting
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service s6rc-fdholder successfully started
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service trt-model-prepare: starting
s6-rc: info: service log-prepare: starting
s6-rc: info: service log-prepare successfully started
s6-rc: info: service nginx-log: starting
s6-rc: info: service go2rtc-log: starting
s6-rc: info: service frigate-log: starting
s6-rc: info: service certsync-log: starting
Generating the following TRT Models: yolov7-320
Downloading yolo weights
s6-rc: info: service go2rtc-log successfully started
s6-rc: info: service go2rtc: starting
s6-rc: info: service nginx-log successfully started
s6-rc: info: service frigate-log successfully started
s6-rc: info: service certsync-log successfully started
s6-rc: info: service go2rtc successfully started
s6-rc: info: service go2rtc-healthcheck: starting
s6-rc: info: service go2rtc-healthcheck successfully started
2024-11-01 18:57:28.037294021  [INFO] Preparing new go2rtc config...
2024-11-01 18:57:28.264011729  [INFO] Starting go2rtc...
2024-11-01 18:57:28.323294212  18:57:28.323 INF go2rtc platform=linux/amd64 revision=b2399f3 version=1.9.2
2024-11-01 18:57:28.323455586  18:57:28.323 INF config path=/dev/shm/go2rtc.yaml
2024-11-01 18:57:28.323724544  18:57:28.323 INF [api] listen addr=:1984
2024-11-01 18:57:28.323769179  18:57:28.323 INF [rtsp] listen addr=:8554
2024-11-01 18:57:28.323920645  18:57:28.323 INF [webrtc] listen addr=:8555/tcp

Creating yolov7-320.cfg and yolov7-320.weights

Done.
2024-11-01 18:57:38.034887147  [INFO] Starting go2rtc healthcheck service...

Generating yolov7-320.trt. This may take a few minutes.

/usr/local/src/tensorrt_demos/yolo/onnx_to_tensorrt.py:149: DeprecationWarning: Use network created with NetworkDefinitionCreationFlag::EXPLICIT_BATCH flag instead.
  builder.max_batch_size = MAX_BATCH_SIZE
/usr/local/src/tensorrt_demos/yolo/onnx_to_tensorrt.py:151: DeprecationWarning: Use set_memory_pool_limit instead.
  config.max_workspace_size = 1 << 30
/usr/local/src/tensorrt_demos/yolo/onnx_to_tensorrt.py:174: DeprecationWarning: Use build_serialized_network instead.
  engine = builder.build_engine(network, config)
`


And the docker-compose.yml file:
`services:
  frigate:
    image: ghcr.io/blakeblackshear/frigate:0.14.0-tensorrt
#    image: ghcr.io/blakeblackshear/frigate:stable
    #    image: ghcr.io/blakeblackshear/frigate:0.14.0-beta4
    container_name: frigate
    restart: always
    privileged: true
    shm_size: "1g"
    deploy:
      resources:
        reservations:
          devices:
            - driver: nvidia
              count: 1
              capabilities: [gpu]
    tmpfs:
      - /tmp/cache:size=8000000000
    ports:
      - "5000:5000"
      - "1935:1935"
      - "8554:8554"
      - "8555:8555/udp"
    volumes:
      # HOST-MACHINE:CONTAINER-machine
      - ./config.yml:/config/config.yml
      - /mnt/storage:/media/frigate   # Added line to map the storage
      - ./db:/config/db               # Database mapped
    environment:
      - NVIDIA_VISIBLE_DEVICES=all
`

What am I missing?

**Reply by NickM-27** [reply]
> /config.yml:/config/config.yml

You're still only mapping the config file which is wrong

**Reply by AJCastroChandri** [reply]
You are right! I replied below.  Thank you Nick!

**Comment by AJCastroChandri** [comment]
I wanted to update:
I correctly mapped the whole config inside docker-compose.yml with:
`      - ./:/config/
`

And it is working correctly now.

Thank you!
